demo/e2e-bdd/shared/helpers/wd-helper.ts

```diff
@@ -0,0 +1,141 @@
+import { browser, by, element, ElementFinder, ExpectedConditions as EC } from 'protractor';
+import { AccordionPo } from '../../pages/accordion.po';
+import { LandingPo } from '../../pages/landing.po';
+import { GettingStartedPo } from '../../pages/gettingStarted.po';
+import { AlertPo } from '../../pages/alert.po';
+import { ButtonsPo } from '../../pages/buttons.po';
+import { CarouselPo } from '../../pages/carousel.po';
+import { CollapsePo } from '../../pages/collapse.po';
+import { DatepickerPo } from '../../pages/datepicker.po';
+import { DropdownsPo } from '../../pages/dropdowns.po';
+import { ModalsPo } from '../../pages/modals.po';
+import { PaginationPo } from '../../pages/pagination.po';
+import { PopoverPo } from '../../pages/popover.po';
+import { ProgressbarPo } from '../../pages/progressbar.po';
+import { RatingPo } from '../../pages/rating.po';
+import { SortablePo } from '../../pages/sortable.po';
+import { TabsPo } from '../../pages/tabs.po';
+import { TimepickerPo } from '../../pages/timepicker.po';
+import { TooltipPo } from '../../pages/tooltip.po';
+import { TypeaheadPo } from '../../pages/typeahead.po';
+
+const TIMEOUT = 6000;
+let message: string;
+
+export async function clickOnElemWithText(selectorToCompare: string, template: string) {
+  const searchedElement = await element(by.cssContainingText(selectorToCompare, template));
+
+  await safeClick(searchedElement);
+}
+
+export async function safeClick(button: ElementFinder) {
```

Comment:
this function does more than one thing. It's better to create another function clickWithScroll() and in this function make scroll and then call safeClick()